packages/ddg2dnr/lib/utils.js

```diff
@@ -170,8 +169,14 @@ function storeInLookup (lookup, key, values) {
 
 /**
  * Generates a declarativeNetRequest rule with the given details.
+ *
+ * @overload
+ * @param {Omit<generateDNRRuleDetails, 'id'>} ruleDetails
+ * @returns {DNRRule}
+ *
+ * @overload
  * @param {generateDNRRuleDetails} ruleDetails
- * @return {DNRRule}
+ * @returns {DNRRuleWithID}
  */
```

Comment:
Is the `DNRRule` in this function the only remaining one left? (I've not checked).

Reply:
No, some folks decided not to use `generateDNRRule()` so the rule types are scattered around unfortunately.

Reply:
I'd consider filing that as a issue to clean up, and migrating `generateDNRRule` to be a class constructor; that way the type def can go away also.